### ISSUE TITLE
Fix S4158 FN: Collection Expressions (C# 12)

### ISSUE THREAD

**Issue by zsolt-kolbay-sonarsource** [SonarSource/sonar-dotnet#8539]
### Description

Collection expressions are not handled by the SE engine, leading to FNs like this:

### Repro steps

```cs
List<int> emptyList = [];
emptyList.Clear();                  // FN

ReadOnlySpan<int> emptySpan = [.. emptyList];
foreach (int element in emptySpan)  // FN
{
    Console.WriteLine(element);
}
```

### Expected behavior

S4158 raises a warning.

### Actual behavior

No S4158 warnings are raised.

### Related information

* C#/VB.NET Plugins version: 9.17
* Visual Studio version: Visual Studio 2022 (17.8.4)
* .NET version: 8.0
* Operating System: Windows 10

### Implementation
With version 4.9.2 of [Microsoft.CodeAnalysis.CSharp](https://www.nuget.org/packages/Microsoft.CodeAnalysis.CSharp/) collection expressions are supported on the CFG level (previously it was always a NoneOperation).
A new `Processor` class is needed in the SE engine that can handle `ICollectionExpressionOperation` types (and also for `ISpreadOperation`).
